<details><summary>Failed to submit graph for 8f59b4a6465bf9f533d84bfbed5ccc14baba975b
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpkseyivcb/onnxruntime-feedstock.yaml --custom-metadata-file /tmp/tmpkseyivcb/custom_metadata.yaml --release-name onnxruntime-feedstock-9-8f59b4a6465bf9f533d84bfbed5ccc14baba975b onnxruntime-feedstock/recipe', with error: 'INFO:root:Found feedstock onnxruntime-feedstock at /tmp/tmpkseyivcb/onnxruntime-feedstock/recipe
INFO:root:Build/Test graph from onnxruntime-feedstock
Created by sbx_user1051 on 2026-03-25 21:17:14
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpkseyivcb/onnxruntime-feedstock.yaml --custom-metadata-file /tmp/tmpkseyivcb/custom_metadata.yaml --release-name onnxruntime-feedstock-9-8f59b4a6465bf9f533d84bfbed5ccc14baba975b onnxruntime-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmpkseyivcb/onnxruntime-feedstock
INFO:root:Creating graph from feedstock at: onnxruntime-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.12 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 5 unchanging vars:(490), changing vars (5): ['blas_impl', 'cuda_compiler_version', 'ep_variant', 'python', 'suffix']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_3u1d0st_.
WARNING: failed to get package records, retrying.  exception was: Unsatisfiable dependencies for platform linux-64: {MatchSpec("libcublas-dev==13.1.0.3=h7354ed3_0"), MatchSpec("cuda-version[version='>=13.0,<13.1.0a0']")}
Encountered problems while solving:
  - package libcublas-dev-13.1.0.3-h7354ed3_0 requires cuda-version >=13.0,<13.1.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ cuda-version >=13.1,<14 * is requested and can be installed;
└─ libcublas-dev =13.1 * is not installable because it requires
   └─ cuda-version >=13.0,<13.1.0a0 *, which conflicts with any installable versions previously reported.
ERROR: Failed to get package records, max retries exceeded.
WARNING: Returning non-final recipe for onnxruntime-cpp-1.24.4-h1234567_0_cuda; one or more dependencies was unsatisfiable:
WARNING: Host: libcublas-dev, cuda-version
WARNING: Host: libcufft-dev, cuda-version
WARNING: Returning non-final recipe for onnxruntime-novec-cpp-1.24.4-h1234567_0_cuda; one or more dependencies was unsatisfiable:
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.expand_variants:Elapsed time for api.render: 271.62 seconds
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-aarch64.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 416, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 643, in from_feedstock
    build_configs, test_configs = expand_build_matrix(
                                  ~~~~~~~~~~~~~~~~~~~^
        feedstock.recipe_dir_path, src_channels, parsed_user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 82, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels, user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 515, in variants_for_feedstock_and_platform
    variants = get_package_variants(parsed.recipe, config, variants=user_variants)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/variants.py", line 701, in get_package_variants
    combined_spec, specs = get_package_combined_spec(
                           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipedir_or_metadata, config=config, variants=variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/variants.py", line 665, in get_package_combined_spec
    validate_spec(f, spec)
    ~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/variants.py", line 204, in validate_spec
    raise ValueError(
        "Variant configuration errors in {}:\n{}".format(src, "\n".join(errors))
    )
ValueError: Variant configuration errors in /tmp/tmpkseyivcb/onnxruntime-feedstock/recipe/conda_build_config.yaml:
  zip fields in zip_key group frozenset({'cuda_compiler_version', 'ep_variant'}) are not all the same length
'
aws lambda request id: 44d61742-e1d2-503d-8bac-757cb2c422bf
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%6044d61742-e1d2-503d-8bac-757cb2c422bf%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/recipe/conda_build_config.yaml
+++ b/recipe/conda_build_config.yaml
@@ -8,31 +8,31 @@ c_stdlib_version:          # [osx]
   - 13.3                   # [osx]
 
 # CUDA 13 CCCL headers require vs2022; CUDA 12 and CPU use vs2019
-c_compiler:                # [win]
-  - vs2019                 # [win]
-  - vs2019                 # [win]
-  - vs2022                 # [win]
-cxx_compiler:              # [win]
-  - vs2019                 # [win]
-  - vs2019                 # [win]
-  - vs2022                 # [win]
+c_compiler:                        # [win]
+  - vs2019                         # [win]
+  - vs2019                         # [win]
+  - vs2022                         # [win]
+cxx_compiler:                      # [win]
+  - vs2019                         # [win]
+  - vs2019                         # [win]
+  - vs2022                         # [win]
 
 cuda_compiler_version:
+  - 12.9                           # [linux64 or win64]
   - 12.9
-  - 12.9
-  - 13.1
+  - 13.1                           # [linux64 or win64]
 
 suffix:
   - ""
   - "-novec"
 ep_variant:
-  - "cuda"                 # [linux64 or win64]
+  - "cuda"                         # [linux64 or win64]
   - "cpu"
-  - "cuda"                 # [linux64 or win64]
+  - "cuda"                         # [linux64 or win64]
 
-zip_keys:                  # [win or linux]
-  -                        # [win or linux]
-    - cuda_compiler_version  # [win or linux]
-    - ep_variant             # [win or linux]
-    - c_compiler             # [win]
-    - cxx_compiler           # [win]
+zip_keys:                          # [linux64 or win64]
+  -                                # [linux64 or win64]
+    - cuda_compiler_version        # [linux64 or win64]
+    - ep_variant                   # [linux64 or win64]
+    - c_compiler                   # [win]
+    - cxx_compiler                 # [win]

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -34,18 +34,18 @@ requirements:
   host:
     - gtest
     - gmock
-    # GPU requirements
+    # GPU requirements — pin toolkit packages to cuda_compiler_version,
+    # let solver pick library versions via cuda-version constraint
     - cuda-nvcc {{ cuda_compiler_version }}        # [(ep_variant == "cuda")]
-    - cudnn {{ cudnn }}                            # [(ep_variant == "cuda")]
     - cuda-version {{ cuda_compiler_version }}     # [(ep_variant == "cuda")]
-    - libcublas-dev {{ cuda_compiler_version }}    # [(ep_variant == "cuda")]
-    - libcurand-dev {{ libcurand }}                # [(ep_variant == "cuda")]
-    - libcusparse-dev {{ libcusparse }}            # [(ep_variant == "cuda")]
-    - libcufft-dev {{ libcufft }}                  # [(ep_variant == "cuda")]
     - cuda-cudart-dev {{ cuda_compiler_version }}  # [(ep_variant == "cuda")]
     - cuda-nvrtc-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
-    # Needed for CUDA profiling
     - cuda-cupti-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
+    - cudnn                                        # [(ep_variant == "cuda")]
+    - libcublas-dev                                # [(ep_variant == "cuda")]
+    - libcurand-dev                                # [(ep_variant == "cuda")]
+    - libcusparse-dev                              # [(ep_variant == "cuda")]
+    - libcufft-dev                                 # [(ep_variant == "cuda")]
     - pip
     - python
     - setuptools
@@ -117,11 +117,11 @@ outputs:
         - {{ compiler('cxx') }}
         - {{ compiler('cuda') }}  # [(ep_variant == "cuda")]
       host:
-        - libcudnn-dev {{ cudnn }}                     # [(ep_variant == "cuda")]
-        - libcufft-dev {{ libcufft }}                  # [(ep_variant == "cuda")]
-        - libcublas-dev {{ cuda_compiler_version }}    # [(ep_variant == "cuda")]
-        - cuda-cupti-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
         - cuda-cudart-dev {{ cuda_compiler_version }}  # [(ep_variant == "cuda")]
+        - cuda-cupti-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
+        - libcudnn-dev                                 # [(ep_variant == "cuda")]
+        - libcufft-dev                                 # [(ep_variant == "cuda")]
+        - libcublas-dev                                # [(ep_variant == "cuda")]
       run_constrained:
         - onnxruntime <0a0       # [suffix == "-novec"]
     test:                        # [(ep_variant == "cpu")]

